src/components/Connection.js

```diff
@@ -39,13 +49,15 @@ export class Connection extends Component {
 
   constructor(props) {
     super(props)
-    this.state = { text: '' }
+    this.state = { hostname: '', port: '3000' }
   }
 
   connect() {
-    const host = this.state.text.trim()
+    const host = this.state.host.trim()
```

Comment:
It was a triple fail, I still had `setState({ text })` too.
I was testing with a default value set in the constructor to avoid re-typing the hostname all the time and I broke it quite brillantly ^^'